### PR TITLE
V1.13.0

### PR DESCRIPTION
**Pull request by kagg-design**

### CODE REVIEW

**Comment by kagg-design** [comment]
@e271828- Please review and merge.

File: src/php/includes/common/request.php [hunk 1]
```diff
@@ -5,6 +5,51 @@
...
+		}
+	}
+
+	if ( ! $client_ip ) {
+		return false;
+	}
+
+	if ( '0.0.0.0' === $client_ip || '::' === $client_ip ) {
+		return false;
+	}
```
Comment (e271828-) [inline src/php/includes/common/request.php:48]: Might want to check for local IPs here: 

```
filter_var(
    $client_ip, 
    FILTER_VALIDATE_IP, 
    FILTER_FLAG_NO_PRIV_RANGE |  FILTER_FLAG_NO_RES_RANGE
)
```

File: src/php/includes/common/request.php [hunk 2]
```diff
@@ -5,6 +5,51 @@
...
+ * Based on the code of the \WP_Community_Events::get_unsafe_client_ip.
+ *
+ * @return false|string
+ */
+function hcap_get_user_ip() {
+	$client_ip = false;
+
+	// In order of preference, with the best ones for this purpose first.
+	$address_headers = [
+		'HTTP_CLIENT_IP',
```
Comment (e271828-) [inline src/php/includes/common/request.php:21]:
```suggestion
                'HTTP_CF_CONNECTING_IP',
		'HTTP_CLIENT_IP',
```

Comment (kagg-design) [reply src/php/includes/common/request.php:21]: The code is based on WP function `\WP_Community_Events::get_unsafe_client_ip`. I have tested it with Cloudflare, it works. Cloudflare provides `HTTP_X_FORWARDED_FOR`.

Comment (kagg-design) [reply src/php/includes/common/request.php:21]: I will add `HTTP_CF_CONNECTING_IP` just for a case...

File: src/php/includes/common/request.php (hunk 1 above)
Comment (kagg-design) [reply src/php/includes/common/request.php]: Thank you, added!

**Review by e271828- — APPROVED** [review]

**Comment by e271828-** [comment]
@kagg-design it occurs to me that the best design from a security perspective would be to let the user select the header at config time, although in practice there's not much benefit to spoofing the input value. Perhaps in a future release.

**Pull request merged**